src/stash/consignment.rs

```diff
@@ -194,6 +194,40 @@ impl Consignment {
         Validator::validate(schema, self, resolver)
     }
 
+    pub fn finalize(
+        &mut self,
+        expose: &BTreeSet<SealEndpoint>,
+        contract_id: ContractId,
+    ) -> usize {
+        let concealed_endpoints =
+            expose.iter().map(SealEndpoint::commit_conceal).collect();
+
+        let mut count = self.state_transitions.iter_mut().fold(
+            0usize,
+            |count, (anchor, transition)| {
+                count
+                    + anchor.conceal_except(contract_id)
+                    + transition.conceal_state_except(&concealed_endpoints)
+            },
+        );
+
+        count =
+            self.state_extensions
+                .iter_mut()
+                .fold(count, |count, extension| {
+                    count + extension.conceal_state_except(&concealed_endpoints)
+                });
+
+        self.endpoints = self
+            .endpoints
+            .clone()
+            .into_iter()
+            .filter(|(_, endpoint)| expose.contains(endpoint))
```

Comment:
Are we deleting the remaining endpoints other than `exposed` for the same reason as above? as they will never be required by receiving party?

Reply:
Yes, these endpoints are endpoints for the sender and we do not want the receiver to know about them